testsuite/test-runner/src/main/java/org/wildfly/core/testrunner/Server.java

```diff
@@ -221,6 +256,66 @@ private void safeCloseClient() {
     }
 
     /**
+     * Reload current server
+     * This method makes sure client on server is still operational after reload is done.
+     * @param adminOnly tells server to boot in admin only mode or normal
+     * @param timeout time in miliseconds to wait for server to come back after reload
+     */
+    public void reload(boolean adminOnly,int timeout){
+        executeReload(adminOnly);
+        waitForLiveServerToReload(timeout); //30 seconds
+    }
+
+    private void executeReload(boolean adminOnly) {
+        ModelNode operation = new ModelNode();
+        operation.get(OP_ADDR).setEmptyList();
+        operation.get(OP).set("reload");
+        operation.get("admin-only").set(adminOnly);
+        try {
+            ModelNode result = client.getControllerClient().execute(operation);
+            Assert.assertEquals("success", result.get(ClientConstants.OUTCOME).asString());
+        } catch (IOException e) {
+            final Throwable cause = e.getCause();
+            if (!(cause instanceof ExecutionException) && !(cause instanceof CancellationException) && !(cause instanceof SocketException) ) {
+                throw new RuntimeException(e);
+            } // else ignore, this might happen if the channel gets closed before we got the response
+        }finally {
+            safeCloseClient();//close existing client
+        }
+    }
+
+    private void recreateClient(){
+        safeCloseClient();
+        client = createClient();
+    }
+
+    protected void waitForLiveServerToReload(int timeout) {
```

Comment:
Use org.jboss.as.test.integration.management.util.ServerReload, which has been ported from full to core.